### DISCUSSION TITLE
method as function parameter

### DISCUSSION THREAD

**Discussion by muescha** [Hammerspoon/hammerspoon#3493] [General]
i like to do some stuff with a method of an object - in this case `hs.hotkey:disable()`

```lua
local hotkeyhandler = hs.hotkey.bind(...)

function myFunction(fn)
	print("myFunction start")
	fn()
	print("myFunction end")
end

-- working
myFunction(function() hotkeyhandler:disable() end)

-- not working - i get an error
myFunction(hotkeyhandler:disable)

-- not working - i get an error
myFunction(hotkeyhandler.disable)
```

where is my mistake so i can not call it with `fn()`?

**Comment by Rhys-T** [comment]
Unfortunately, there's not really a syntax in Lua for what you're trying to do. Lua doesn't have 'bound methods' like in, say, Python. When you write `foo:bar(baz)`, `foo:bar` by itself doesn't really exist - the colon is as much a part of the call syntax as the parentheses are!

If you want something shorter than writing an anonymous function wrapper each time, you can write a function to generate those wrappers for you:

```lua
-- 'M' for 'method'. You can name this whatever you want.
function M(object, name)
	local func = object[name]
	return function(...)
		return func(object, ...)
	end
end

myFunction(M(hotkeyhandler, 'disable'))
```

**Comment by cmsj** [comment]
Marking as closed because @Rhys-T is correct and wrapping the method call is the correct choice here.
